src/Workspaces/Remote/ServiceHub/Shared/ServiceHubServiceBase.cs

```diff
@@ -46,6 +47,26 @@ protected ServiceHubServiceBase(Stream stream, IServiceProvider serviceProvider)
             Rpc.Disconnected += OnRpcDisconnected;
         }
 
+        protected ServiceHubServiceBase(IServiceProvider serviceProvider, Stream stream)
```

Comment:
behavior of these 2 ctor is different. existing one start listening stream right away but new one only does that when derived type call start listening.